### PR TITLE
Make data_suspect usage optional

### PR DESCRIPTION
**Pull request by sratcliffe**
How about something like this ?

### CODE REVIEW

File: scripts/ingest.py [hunk 1]
```diff
@@ -115,6 +115,10 @@ def parse_args() -> argparse.Namespace:
     parser.add_argument(
         '--no-excise', dest='excise', action='store_false',
         help='disable excision of flagged data [default=no]')
+    parser.add_argument(
+        '--use-data-suspect', dest='use_data_suspect', action='store_false',
```
Comment (ludwigschwardt) [inline scripts/ingest.py]: No need to set the destination - the dashes are automatically converted to underscores.

Comment (bmerry) [reply scripts/ingest.py]: Did you mean to use `store_true`? Currently if you specify `--use-data-suspect`, then `use_data_suspect` becomes False.

File: katsdpingest/ingest_session.py [hunk 2]
```diff
@@ -1257,18 +1258,21 @@ def sensor_value(telstate, name):
...
-            input_suspect = {label: input_suspect_sensor[i] for i, label in enumerate(input_labels)}
-        else:
-            input_suspect = {}
+        input_suspect = {}    # type: Dict[str, int]
+        if self.use_data_suspect:
+            input_suspect_sensor = sensor_value(
+                self.telstate_cbf, 'input_data_suspect')
+            if input_suspect_sensor is not None:
+                input_suspect = \
+                    {label: input_suspect_sensor[i] for i, label in enumerate(input_labels)}
```
Comment (bmerry) [inline katsdpingest/ingest_session.py]: Personally I prefer to break the line at the `for` rather than using a `\`.

File: scripts/ingest.py [hunk 3]
```diff
@@ -115,6 +115,10 @@ def parse_args() -> argparse.Namespace:
...
+        help=('use the CAM provided input-data-suspect and channel-data-suspect '
```
Comment (bmerry) [inline scripts/ingest.py]:
```suggestion
        help=('use the CAM-provided input-data-suspect and channel-data-suspect '
```

File: katsdpingest/ingest_session.py [hunk 4]
```diff
@@ -1257,18 +1258,21 @@ def sensor_value(telstate, name):
...
 
         baselines = self.bls_ordering.sdp_bls_ordering
         input_labels = self.cbf_attr['input_labels']
-        input_suspect_sensor = sensor_value(
-            self.telstate_cbf, 'input_data_suspect')
-        if input_suspect_sensor is not None:
-            input_suspect = {label: input_suspect_sensor[i] for i, label in enumerate(input_labels)}
-        else:
-            input_suspect = {}
+        input_suspect = {}    # type: Dict[str, int]
```
Comment (bmerry) [inline katsdpingest/ingest_session.py]: I think it's `bool`, not `int`.

**Pull request merged**